### PR TITLE
Hiding and showing Emojis in the UI!

### PR DESCRIPTION
**Pull request by heathervv**
I think I've got this working nicely now! Take a look and let me know what you think

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -101,7 +101,7 @@
         <li class="emojis--item"><a href="#" title="speak_no_evil">🙊</a></li>
         <li class="emojis--item"><a href="#" title="skull">💀</a></li>
         <li class="emojis--item"><a href="#" title="alien">👽</a></li>
-        <li class="emojis--item"><a href="#" title="poo">💩</a></li>
+        <li class="emojis--item"><a href="#" title="poop">💩</a></li>
```
Comment (heathervv) [inline index.html:104]: Updated to match `emoji.js`

File: main.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // DOM utility functions
 const addClass = (el, newClass) => {
-  el.className += ` ${newClass}`;
+  el.classList.add(newClass);
```
Comment (heathervv) [inline main.js:3]: `classList` only adds the class to the element if it isn't already on there.

File: main.js [hunk 3]
```diff
@@ -16,26 +16,54 @@ const removeClass = (el, oldClass) => {
...
   const TYPE_INTERVAL = 10000;
+  const siteContainer = document.getElementsByTagName('body');
 
   inputField.addEventListener("input", (event) => {
     const value = event.target.value;
     if (typeTimeout) window.clearTimeout(typeTimeout);
 
     if (value) {
       typeTimeout = delay(matchKeyword(value), TYPE_INTERVAL);
+    } else {
```
Comment (heathervv) [inline main.js:30]: This entire else is where I reset the search and show all emojis again

File: main.js [hunk 4]
```diff
@@ -16,26 +16,54 @@ const removeClass = (el, oldClass) => {
...
     if (matchedObjects.length == 0) return;
 
     matchedObjects.map((obj) => {
       matchedEmojiTitles.push(obj.keywords[0]);
     });
 
-    // do something with matchedEmojiTitles here
+    const prevFilter = document.getElementsByClassName('visible');
+
+    oldEmojis(prevFilter);
```
Comment (heathervv) [inline main.js]: Every time you search I remove the last search's results!

File: main.js [hunk 5]
```diff
@@ -16,26 +16,54 @@ const removeClass = (el, oldClass) => {
...
+
+
+    for (const emoji of matchedEmojiTitles) {
+      const emojiItem = document.querySelector('[title="' + emoji + '"]');
+
+      if (emojiItem) {
+        addClass(emojiItem.parentElement,'visible');
+      }
+    }
```
Comment (heathervv) [inline main.js:51]: Here is where the magic is happening :sparkles: I'm iterating through the array you supplied and comparing against the titles of the emojis in the markup.

File: main.js [hunk 6]
```diff
@@ -16,26 +16,54 @@ const removeClass = (el, oldClass) => {
...
+
+    addClass(siteContainer[0], 'filtering');
+  }
+
+  const oldEmojis = (value) => {
+    for (const prev of value) {
+      removeClass(prev, 'visible');
+    };
```
Comment (heathervv) [inline main.js:59]: I turned this into a function because I use it in two places (line 32 and 49)

File: main.js (hunk 5 above)
Comment (skimke) [inline main.js:51]: Just curious - have you tried declaring a variable where you can store the query string?
Like:
`const query = '[title="${emoji}"]'`
`const emojiItem - document.querySelector(query);`
I guess this is much neater and in one line though. 😞

File: main.js (hunk 6 above)
Comment (skimke) [inline main.js:59]: Not-A-Big-Deal-Style-Thing: Could we get rid of the extra line space on line 50?

More-Importantly: I'm wondering if we could make the function's name more descriptive and based on actions - something like `resetOldEmojis` or `revealEmojis`.

File: index.html (hunk 1 above)
Comment (skimke) [inline index.html:104]: Haha, we could add `poo` to the keywords, I guess

File: main.js (hunk 5 above)
Comment (heathervv) [inline main.js:51]: Good idea! 

I'm looking into this right now, and for some reason even setting it as a string isn't working. Like when I use the exact code you have above and then console.log() query I just get `[title="${emoji}"]`

... weird, but I'm going to look into it!

File: main.js (hunk 3 above)
Comment (skimke) [inline main.js:30]: This works well. I noticed though that if I take my time typing something, there's slight bugginess (more results than expected). Want to kick up the `TYPE_INTERVAL`? Have you noticed it as well?

Comment (heathervv) [inline main.js:30]: I've DEFINITELY noticed it. I'll play with increasing/decreasing `TYPE_INTERVAL` and see if that resolves the issue.

File: main.js (hunk 5 above)
Comment (heathervv) [inline main.js:51]: Found the issue! I was using single quotes like you would for ES5 strings - but you need to use the back tick character instead.

File: main.js [hunk 7]
```diff
@@ -16,26 +16,47 @@ const removeClass = (el, oldClass) => {
...
   let typeTimeout;
   const TYPE_INTERVAL = 10000;
+  const siteContainer = document.getElementsByTagName('body');
 
   inputField.addEventListener("input", (event) => {
     const value = event.target.value;
+    const prevFilter = document.getElementsByClassName('visible');
     if (typeTimeout) window.clearTimeout(typeTimeout);
+  
+    resetOldEmojis(prevFilter);
```
Comment (heathervv) [inline main.js:26]: @skimke So I think I've resolved the "emojis not related to the search" issue without editing `TYPE_INTERVAL`

Instead I moved the `resetOldEmojis` function to trigger anytime the input field is changed where before it was triggering if the input was empty and when we were trying to implement the search. This seems to have resolved the issue! Let me know if you're still seeing it.

File: _/scss/base/_layout.scss [hunk 8]
```diff
@@ -2,15 +2,19 @@
...
 section {
-	max-width: 1000px;
-	margin: 0 auto;
-	padding: 0 50px;
+    max-width: 1000px;
+    margin: 0 auto;
+    padding: 0 50px;
+}
+
+.hidden {
```
Comment (heathervv) [inline _/scss/base/_layout.scss:18]: @skimke Added a global hidden class for you to use on the inputs!

**Comment by skimke** [comment]
This is great! Thanks @heathervv 🚀

**Pull request merged**